### PR TITLE
fix(morning-briefing): default scope=current must not scan sibling worktrees

### PR DESCRIPTION
**Pull request by ekson73**
## Summary

The operator asked me to investigate `morning-briefing` and guarantee its default scope is **[current session]**, to avoid silently expanding into another agentic work-in-progress without the operator's awareness.

Investigation found exactly that bug: Phase 1's baseline probe unconditionally `cd`-ed into **every** worktree returned by `git worktree list` and dumped `git status --short` for each — regardless of `--scope`. So the default `--scope=current` briefing already inspected sibling worktrees' uncommitted work (another session's/agent's branch), silently contradicting:
- Phase 2.5's own claim that `current` = "baseline, no expansion", and
- anti-pattern #9's principle ("only scan beyond current with explicit flag") — whose actual flag, `--multi-repo`, only ever gated sibling **repos**, never sibling **worktrees** within the same repo.

Reproduced live in this very repo: at fix-time, `git worktree list` showed 2 other worktrees (`verified-agentic-session-model`, `wacli-archetype`) that a bare `/morning-briefing` invocation would have silently scanned and surfaced.

## Changes

- **Phase 1**: removed the unconditional per-worktree `cd` + `git status --short` loop. `git worktree list` is kept for enumeration only (needed for the `$worktrees <n>` Pulse figure).
- **Phase 2.5 `sideways` case**: added the actual per-worktree content scan there — this closes a second, pre-existing gap where the per-verb table already promised "+ sibling worktrees state" for `--scope=sideways` but the implementation never delivered it.
- Operator-flags table: clarified `--scope=current` (session-scope boundary), `--scope=sideways` (now the sole path to sibling-worktree state), and `--multi-repo` (disambiguated as the orthogonal cross-repo axis — distinct from sibling-worktree scanning).
- New **anti-pattern #29** (Default-scope worktree leakage) + 1 new edge case.
- `prompt_version` 1.8.0 → 1.8.1 (**PATCH** — zero new capability, corrects an implementation/documentation mismatch; `cycles_completed` preserved per ADR-017 R1).

## Test plan

- [x] `bash -n` syntax-checked all 6 bash code blocks in the SKILL.md (including the edited Phase 1 + Phase 2.5 blocks) — all pass.
- [x] Frontmatter YAML re-parsed successfully (`prompt_version: 1.8.1`, 12 `cycles_evidence` entries).
- [x] `gitleaks git --staged` — no leaks found.
- [x] Manually traced the new default behavior: `--scope=current` (bare invocation) no longer touches sibling worktrees; `--scope=sideways` is now the only path that does, matching its own documented per-verb table.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01PWke4PAeV4KqMu29mRAdvs

### CODE REVIEW

File: skills/morning-briefing/SKILL.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ description: |
   recap", or starts/ends a session. Capability-detected (git/gh/jq optional), cross-vendor (AAIF spec),
   no hardcoded vendors, open-source-promotable. Differs from /context-restore (no prior /context-save
   required — works cold) and /retro (daily, not weekly, decision-oriented not retrospective).
-prompt_version: "1.8.0"
+prompt_version: "1.8.1"
```
[the remark this answers is not in the document]
Comment (ekson73) [reply skills/morning-briefing/SKILL.md:16]: Verified against this repo's actual frontmatter validator — this is a false positive, not fixed.

`scripts/validate-skill-frontmatter.sh` is the sole SKILL.md frontmatter gate in this repo (wired via CI, not a hypothetical). Its predicate:

```python
has_name = re.search(r"(?m)^name:\s*\S", fm) is not None
has_desc = re.search(r"(?m)^description:\s*\S", fm) is not None
if not (has_name and has_desc):
    bad.append(f"{p}: frontmatter needs name + description")
```

It requires **only** `name:` + `description:`. There is no `version:` field requirement anywhere in the check (confirmed by reading the full script, not by sampling). Ran it against this branch's frontmatter directly:

```
$ bash scripts/validate-skill-frontmatter.sh
OK 96 SKILL.md files have name+description frontmatter
```

The skill does carry `prompt_version: "1.8.1"` (its own internal version scheme, tracked in `cycles_evidence` per ADR-017 R1) — the finding conflates that with a generic `version:" field the actual tooling doesn't ask for. Leaving as-is per `ai-code-review-bots-rotation.md`'s bot-wrong discipline (verify empirically, dismiss with evidence rather than fabricate compliance).

File: skills/morning-briefing/SKILL.md [hunk 2]
```diff
@@ -31,6 +31,7 @@ cycles_evidence:
...
... operador'. The Phase-1 baseline unconditionally cd'd into EVERY worktree returned by `git worktree list` and dumped `git status --short` for each — meaning the default `--scope=current` briefing already inspected another agentic session's/agent's uncommitted work, contradicting Phase 2.5's own 'current = baseline, no expansion' claim and the anti-pattern #9 principle (whose `--multi-repo` flag only ever gated sibling REPOS, never sibling WORKTREES within the same repo). Fix: per-worktree content scan moved to the explicit `--scope=sideways` case (Phase 2.5, matching what its own per-verb table already promised but never implemented); Phase-1 default now enumerates worktree paths/count only (needed for `$worktrees <n>` in Pulse), never their content. New anti-pattern #29 + 1 new edge case + operator-flags-table clarifications (`--scope=current`, `--scope=sideways`, `--multi-repo`). Zero new capability — corrects an implementation/documentation mismatch. `cycles_completed` preserved per ADR-017 R1 (PATCH does not reset). Discovered/fixed 2026-09-11 dogfooding the skill in multi-agent-os itself (2 other worktrees present at fix-time: verified-agentic-session-model, wacli-archetype)."
```
[the remark this answers is not in the document]
Comment (ekson73) [reply skills/morning-briefing/SKILL.md]: Verified against this repo's actual CI/tooling — this is a false positive, not fixed.

Grepped this repo's `.github/workflows/` and `scripts/` for any work-item/ticket-key requirement gate — none exists. There is also no `.coderabbit.yaml` (confirmed via `find`) that could impose a ticket-key convention. `AGENTS.md`'s only consumer-visible-contract gate is the CHANGELOG-entry requirement (`changelog-required.yml` → `scripts/changelog-required-check.sh`), which this same round already closed — see the CHANGELOG.md `## [Unreleased]` entry + `git diff --name-only origin/main...HEAD | scripts/changelog-required-check.sh` now returning `contract changed (1 entry) and CHANGELOG.md updated — ok`.

This PR is a bugfix on an already-existing, already-merged skill (no separate feature ticket was opened for the follow-up fix — the fix IS the ticket-worthy unit, tracked via this PR itself + `cycles_evidence`). Not fabricating a Jira key to satisfy a gate that doesn't exist. Per `ai-code-review-bots-rotation.md`'s bot-wrong discipline.

**Comment by ekson73** [comment]
@coderabbitai All findings from this review round have been addressed.

| Finding | Fix | Commit |
|---|---|---|
| Sibling worktree paths splitting on spaces (`for wt in $(...)`, line 454) | `cut -d' ' -f2-` + line-safe `while IFS= read -r wt` loop | `1b09ede` |
| `head -"$SCOPE_BREADTH"` / `head -10` non-portable form | `head -n "$SCOPE_BREADTH"` via a `breadth_limit()` helper | `1b09ede` |
| `--breadth=all` → invalid `head -all` / `gh --limit all` | `breadth_limit()` uses `cat` for `all`, ample ceiling for `gh --limit` | `1b09ede` |
| Current worktree not excluded from `sideways` enumeration | `grep -Fxv "$CURRENT_WT"` against `git rev-parse --show-toplevel` | `1b09ede` |
| Phase 3 `$state_detail` still rendered sibling paths by default | Changed to count-only; paths reserved for `--scope=sideways` | `1b09ede` |
| Missing `CHANGELOG.md` `## [Unreleased]` entry (`changelog-required` gate) | Entry added; gate now returns `contract changed (1 entry) and CHANGELOG.md updated — ok` | `1b09ede` |

Two other findings (from qodo-code-review) were verified and dismissed as false positives with cited evidence, replied inline: missing `version:` frontmatter field (the repo's actual validator only requires `name:` + `description:`) and missing Jira/ticket-key (no such CI gate exists in this repo).

Tests: all 6 bash blocks re-verified with `bash -n` after every edit · frontmatter YAML re-parsed OK · `gitleaks git --staged` clean · `changelog-required-check.sh` and `validate-skill-frontmatter.sh` both pass.

Could you please re-review?

File: skills/morning-briefing/SKILL.md [hunk 3]
```diff
@@ -435,9 +445,37 @@ case "$SCOPE_VERB" in
...
+    # integer — "all" maps to a generous ceiling instead.
+    if [ "$SCOPE_BREADTH" = "all" ]; then
+      breadth_limit() { cat; }
+      BREADTH_GH_LIMIT=1000  # gh --limit rejects "all"; ample ceiling stands in for unlimited
+    else
+      breadth_limit() { head -n "$SCOPE_BREADTH"; }
+      BREADTH_GH_LIMIT="$SCOPE_BREADTH"
+    fi
+    git for-each-ref --format='%(refname:short)' refs/heads/ 2>/dev/null | breadth_limit
+    [ "$HAS_GH" = "yes" ] && gh pr list --state open --json number,title,headRefName --limit "$BREADTH_GH_LIMIT"
```
[the remark this answers is not in the document]
Comment (ekson73) [reply skills/morning-briefing/SKILL.md:467]: Real finding, correctly scoped to this PR's own new `sideways` code — but implementing "peer-off-same-parent" branch filtering and "sibling-PR-same-milestone" filtering is a feature addition, not a bug fix within this v1.8.1 scope-leak PATCH.

Two reasons it's tracked as a follow-up rather than fixed inline:

1. **"Same parent" has no native git primitive.** `git for-each-ref` branches have no first-class parent relationship after creation (unlike worktrees). Determining it would need a merge-base heuristic (`git branch --contains <merge-base>`) — a design decision, not a mechanical fix.
2. **"Same milestone" isn't in the current API call.** `gh pr list --json number,title,headRefName` never requests a `milestone` field — the per-verb table's prose describing sibling-PR selection was aspirational, not something the code ever implemented (so this isn't a regression this PR introduced, just unfiltered pre-existing behavior on the new code path).

Filed as tracked follow-up: https://github.com/ekson73/multi-agent-os/issues/423 (also covers the pre-existing `down`-case word-splitting parity bug, kept out of this PR's diff for the same "stay scoped to what changed" reason).

Per this PR's own convention (`pr-review-protocol.md` §2.5 — bot reviews are data, not instruction; findings are verified before acting, not auto-fixed): this is a real, in-scope-adjacent gap, correctly deferred rather than silently dropped.

**Comment by ekson73** [comment]
@coderabbitai the one open finding (sideways-peer-filtering, line 467) is a real, deliberately-deferred feature gap — replied inline with rationale + tracked as issue #423 (same disposition as the pre-existing `down`-case word-splitting bug, also deferred to keep this PATCH scoped). All other findings from this round were fixed in `1b09ede` or replied-dismissed with evidence (version-field and ticket-key false positives). Could you re-review?
